test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java

```diff
@@ -54,15 +55,22 @@
             {"InitialRAMFraction",        "64"},
             {"TLABStats",                 "false"},
             {"AllowRedefinitionToAddDeleteMethods", "true"},
-            {"UseContainerCpuShares",      "false"},
-            {"PreferContainerQuotaForCPUCount", "true"},
 
             // deprecated alias flags (see also aliased_jvm_flags):
             {"DefaultMaxRAMFraction", "4"},
             {"CreateMinidumpOnCrash", "false"}
           }
         ));
         DEPRECATED_OPTIONS = deprecated.toArray(new String[][]{});
+
+        ArrayList<String[]> deprecated_linux = new ArrayList(
```

Comment:
Seems simpler to just add these to `deprecated` before line 64.

Reply:
So move the `Platform.isLinux()` check up here and optionally
ArrayList.add() the needed strings? That does sound simpler.

Reply:
I changed the code as David suggested.

I also added a comment to explain why we use the clumsy-looking  `new ArrayList(Arrays.asList(new String[][] {...}))` construct. A simple `new String[][] {...}` would suffice if we didn't need to add optional checks for specific platforms.